### PR TITLE
Fix concurrency issue in gps loader

### PR DESCRIPTION
**Pull request by aPlacuzzi**

### CODE REVIEW

File: alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java [hunk 1]
```diff
@@ -58,6 +59,7 @@
         .withNarrowingConversions()
         .withAutoBoxing()
         .build();
+    private static final Semaphore SEMAPHORE = new Semaphore(1);
```
Comment (DanySK) [inline alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java:62]: A semaphore with a single permit is better called `MUTEX`

File: alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java [hunk 2]
```diff
@@ -176,14 +180,17 @@ public TraceLoader(final String path,
...
-    private static GPSTimeAlignment makeNormalizer(final String clazzName, final Object... args) {
+    private static GPSTimeAlignment makeNormalizer(final String clazzName, final Object... args) throws InterruptedException {
         final String fullName = clazzName.contains(".")
                 ? clazzName
                 : GPSTimeAlignment.class.getPackage().getName() + "." + clazzName;
         try {
             final Class<?> targetClass = ResourceLoader.classForName(fullName);
             if (GPSTimeAlignment.class.isAssignableFrom(targetClass)) {
-                return (GPSTimeAlignment) FACTORY.build(targetClass, args).getCreatedObjectOrThrowException();
+                SEMAPHORE.acquire();
```
Comment (DanySK) [inline alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java:190]: Use `acquireUninterruptibly()` and remove the exception propagation

File: alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java [hunk 3]
```diff
@@ -176,14 +180,17 @@ public TraceLoader(final String path,
...
+                var normalizer = (GPSTimeAlignment) FACTORY.build(targetClass, args).getCreatedObjectOrThrowException();
```
Comment (DanySK) [inline alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java:191]: This cast is unchecked

**Review by DanySK — CHANGES_REQUESTED** [review]

File: alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java (hunk 1 above)
Comment (aPlacuzzi) [reply alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java]: done

File: alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java (hunk 2 above)
Comment (aPlacuzzi) [reply alchemist-maps/src/main/java/it/unibo/alchemist/boundary/gpsload/impl/TraceLoader.java]: done

**Review by DanySK — CHANGES_REQUESTED** [review]

**Review by DanySK — APPROVED** [review]

**Pull request merged**